website/docs/plugin/framework/handling-data/attributes.mdx

```diff
@@ -468,6 +468,308 @@ attribute value in a way that Terraform will understand. `attr.Value` has the
+
+  map_attribute = { "map-key-1" : "map-value-1" }
+
+  set_attribute = ["set-element-1", "set-element-2"]
+}
+```
+
+#### Schema
+
+Define collection attributes in the schema as follows:
+
+```go
+func (e *exampleResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+    resp.Schema = schema.Schema{
+        Attributes: map[string]schema.Attribute{
+            "list_attribute": schema.ListAttribute{
+                Optional:    true,
+                ElementType: types.StringType,
+            },
+
+            "map_attribute": schema.MapAttribute{
+                Optional:    true,
+                ElementType: types.StringType,
+            },
+
+            "set_attribute": schema.SetAttribute{
+                Optional:    true,
+                ElementType: types.StringType,
+            },
+        },
+    }
+}
+```
+
+### Object Attribute
+
+Use an object attribute when the attribute is an atomic unit and all fields defined within it need to be specified.
+
+Nested attributes (i.e., ListNestedAttribute, MapNestedAttribute, SetNestedAttribute and SingleNestedAttribute) should be used when any of the inner fields should have their own flags or metadata (e.g., Required, Optional, Sensitive etc).
+
+#### Terraform Configuration
+
+Use the following syntax in Terraform configuration for object attributes:
+
+```hcl
+resource "example_resource" "example" {
+  object_attribute = {
+    bool_attribute    = true
+    float64_attribute = 1234.5
+    int64_attribute   = 9223372036854775807
+    list_attribute    = ["obj-list-element", "obj-list-element"]
+    map_attribute     = { "obj-map-key-1" : "obj-map-value-1" }
+    number_attribute  = 1.79769313486231570814527423731704356798070e+1000
+    set_attribute     = ["obj-set-element-1", "obj-set-element-2"]
+    string_attribute  = "obj-string"
+  }
+}
+```
+
+#### Schema
+
+Define object attributes in the schema as follows:
+
+```go
+func (e *exampleResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+    resp.Schema = schema.Schema{
+        Attributes: map[string]schema.Attribute{
+            "object_attribute": schema.ObjectAttribute{
+                Optional: true,
+                AttributeTypes: map[string]attr.Type{
+                    "bool_attribute":    types.BoolType,
+                    "float64_attribute": types.Float64Type,
+                    "int64_attribute":   types.Int64Type,
+                    "list_attribute":    types.ListType{ElemType: types.StringType},
+                    "map_attribute":     types.MapType{ElemType: types.StringType},
+                    "number_attribute":  types.NumberType,
+                    "set_attribute":     types.ListType{ElemType: types.StringType},
+                    "string_attribute":  types.StringType,
+                },
+            },
+        },
+    }
+}
+```
+
+### Nested Attributes
```

Comment:
I think these are omitted from the above sections, so it'll be good to walk through these. Do you think you could structure this new content to look more similar to the existing content (e.g. split per attribute type). Thanks!

Reply:
Ah ha, it was awkwardly on the Schemas page -- I think that content can be moved here. 👍

Reply:
Have moved the content and consolidated the TF configuration and schema examples for nested attributes.